src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

```diff
@@ -0,0 +1,374 @@
+        string packageInstallationCommand = distroType == DistroType.Deb ? "dpkg -i --force-all" : "rpm -i";
+        bool useAndOperator = false;
+        foreach (string package in rpmPackageList)
+        {
+            sb.AppendLine(" \\");
+            sb.Append($"    {(useAndOperator ? "&&" : "")} {packageInstallationCommand} {package}");
+            useAndOperator = true;
+        }
+        sb.AppendLine("");
+
+        // Set environment for nuget.config
+        sb.AppendLine("");
+        sb.AppendLine("# Set custom nuget.config");
+        sb.AppendLine("ENV RestoreConfigFile=/NuGet.config");
+
+        // Find scenario-tests binary in context/scenario-tests
+        string? scenarioTestsBinary = Directory.GetFiles(Path.Combine(_contextDir, "scenario-tests"), "Microsoft.DotNet.ScenarioTests.SdkTemplateTests.dll", SearchOption.AllDirectories).FirstOrDefault();
+        if (scenarioTestsBinary == null)
+        {
+            throw new Exception("Scenario tests binary not found");
+        }
+        scenarioTestsBinary = scenarioTestsBinary.Replace(_contextDir, "").Replace("\\", "/");
+
+        // Set entry point
+        sb.AppendLine("");
+        sb.AppendLine($"ENTRYPOINT [ \"dotnet\", \"{scenarioTestsBinary}\", \"--dotnet-root\", \"/usr/share/dotnet\" ]");
+
+        string dockerfile = Path.Combine(_contextDir, Path.GetRandomFileName());
+        File.WriteAllText(dockerfile, sb.ToString());
+        return dockerfile;
+    }
+
+    private bool AnyTestFailures(string testResultSummary)
+    {
+        var parts = testResultSummary.Split(',')
+         .Select(part => part.Split(':').Select(p => p.Trim()).ToArray())
+         .Where(p => p.Length == 2)
+         .ToDictionary(p => p[0], p => int.Parse(p[1]));
+
+        return parts["Errors"] > 0 || parts["Failures"] > 0;
+    }
+
+    private void AddPackage(List<string> packageList, string prefix, DistroType distroType)
+    {
+        packageList.Add(Path.GetFileName(GetContentPackage(prefix, distroType)));
+    }
+
+    private string GetContentPackage(string prefix, DistroType distroType)
+    {
+        string matchPattern = DistroType.Deb == distroType ? "*.deb" : "*.rpm";
+        string[] rpmFiles = Directory.GetFiles(_contextDir, prefix + matchPattern, SearchOption.AllDirectories)
+            .Where(p => !Path.GetFileName(p).Contains("dotnet-runtime-deps-"))
+            .ToArray();
+        if (rpmFiles.Length == 0)
+        {
+            throw new Exception($"RPM package with prefix '{prefix}' not found");
+        }
+
+        return rpmFiles.OrderByDescending(f => f).First();
+    }
+
+    private string GetMatchingDepsPackage(string baseImage, DistroType distroType)
+    {
+        string matchPattern = "dotnet-runtime-deps-*.deb";
+        if (distroType == DistroType.Rpm)
+        {
+            string? depsId = null;
+            if (baseImage.Contains("opensuse"))
```

Comment:
Why is this not hardcoded to azurelinux since it appears to be the only exercisable codepath?

Reply:
Why not refactor at that time?  I don't like the thought of adding code that is not certain to be used, it is a form of technical debt.  We don't have runtime-deps images for these distros.  What is the ROI of adding them?